app/src/main/java/nl/tudelft/b_b_w/model/DatabaseHandler.java

```diff
@@ -163,6 +169,48 @@ public int lastSeqNumberOfChain(String owner) {
 
 
     /**
+     * Function to backtrace the contact name given the hash that refer to their block
+     * @param hash hash of the block which owner name we want to find from
+     * @return name of owner
+     */
+    public String getContactName(String hash)
+    {
+
+        SQLiteDatabase db = this.getReadableDatabase();
+        Cursor cursor = db.query(TABLE_NAME,
+                _columns,
+                KEY_OWN_HASH + " = ? ",
+                new String[]{
+                        hash
+                }, null, null, null, null);
+
+        //When returning an exception the whole program crashes,
+        //but we want to preserve the state.
+        if (cursor.getCount() < 1) return null;
+
+        cursor.moveToFirst();
+
+        final String blockType = (cursor.getInt(7) > 0) ?  "REVOKE" : "BLOCK";
+        Block block = BlockFactory.getBlock(
+                blockType,cursor.getString(0),
+                cursor.getString(2),
+                cursor.getString(3),
+                cursor.getString(4),
+                cursor.getString(5),
+                cursor.getString(6),
+                cursor.getInt(7));
+        block.setSeqNumberTo(cursor.getInt(1));
+        db.close();
+        cursor.close();
+
+
+        return (block.getSequenceNumber()==1) ? block.getOwner() : block.getOwner()+"'s friend #" + (block.getSequenceNumber() - 1);
```

Comment:
Do you mean to actually bring the block.setSeqNumberTo() into the BlockFactory.AddBlock?